### ISSUE TITLE
Tests Break on 404 Errors & Doesn' t Continue

### ISSUE THREAD

**Issue by onurtr** [mikestead/lighthouse-batch#68]
Hi;

Again thanks for this useful tool. :) We' ve a problem about broken links. 

We' re running lighthouse-batch with this command:

`lighthouse-batch -f src/urls_all.txt --accessibility 60 --params "--port 49965 --preset=desktop" --out "./lighthouse/desktop/"`

If there are some broken links in the urls_all.txt, lighthouse-batch gives the error below error & breaks down in the middle of the process.

![resim](https://user-images.githubusercontent.com/7816149/127982882-40944653-a914-488e-95cb-d3dff49e746c.png)

For example it successfully tests 1000 pages but got an 404 error in the 1001th page. It doesn' t continue to text the next URLs & because it doesn' t create summary.json file, we need to whole process again just after deleting the broken link from the text file.

Is there any way to continue to run the script even we had broken links in the url list?

Kind regards.

**Comment by onurtr** [comment]
Additional notes, it seems it' s not breaking only for 404 errors. It can break on other errors as well.

**Comment by pyymenta** [comment]
@onurtr I'm having the same problem here:

![image](https://user-images.githubusercontent.com/13206817/128182740-2b245fd6-cb83-4171-bc93-63b1d37b7e5a.png)

```
Distributor ID: Ubuntu
Description:    Ubuntu 21.04
Release:        21.04
Codename:       hirsute

NodeJs: 16.4.0
Npm: 7.18.1
```

If I roll back to version 7.1.0, it works fine:

![image](https://user-images.githubusercontent.com/13206817/128183471-99ecac14-6965-455d-b7e6-7e98b4af4f2f.png)

**Comment by onurtr** [comment]
Hi @pyymenta ;

I need to use 8+ version because there' s a bugfix in that version about file name lenght should be less than 255. I hope we can solve this issue. :)

**Comment by pyymenta** [comment]
@onurtr Absolutely,

What's the nodejs version are you running?

**Comment by onurtr** [comment]
Hi @pyymenta;

I' m running on 14.16.1. 

Kind regards. :)

**Comment by onurtr** [comment]
Hi there;

Is there any news about this topic? :)

Kind regards.

**Comment by mikestead** [comment]
Hey, just on vacation atm, will get to this soon though, apologies for the delay

**Comment by onurtr** [comment]
Hi @mikestead ;

I did this kind of fix on my local:

In checkBudgets function:

if (options.accessibility > 0 **&& summary.detail**) {
if (options.performance > 0 **&& summary.detail**) {

etc.

Hope it was a right fix & can be helpful for you.

**Comment by mikestead** [comment]
Fixed via #69
